### ISSUE TITLE
Add migration for ESM support to Angular CLI/Nx workspaces [1.x]

### ISSUE THREAD

**Issue by brandonroberts** [analogjs/analog#809]
### Which scope/s are relevant/related to the feature request?

create-analog

### Information

In the 1.0 release, the Analog Vite plugins will be published as ESM-only.

For existing Angular CLI workspaces, a few migrations will need to be made.
- Add `"type": "module"` to the root package.json
- Rename any `.js` files to `.cjs`, such as `tailwind.config.js` and `postcss.config.js`
- Update any builders that were previously using `@nx/vite` to use `@analogjs/platform` equivalents.
- Update the `root` property in the `angular.json` to be `"."` instead of `""`

For existing Nx workspaces with an Analog project
- Add a `package.json` in the project root that includes `{ "type": "module" }` so Vite resolves it as ESM
- File renaming may not be required because Nx is still CommonJS, and dynamically imports Vite as ESM

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

**Comment by nckirik** [comment]
the 1.0 release doesn't have `@analogjs/trpc` package.

@brandonroberts mentioned that trpc support will be moved to `@spartan-ng/trpc` that is maintained by @goetzrobin

currently `@spartan-ng/trpc` doesn't have any releases for now i'm keeping `@analogjs/trpc` at 0.2.45.

is this the right way to go?

**Comment by brandonroberts** [comment]
@nckirik yes, that's correct. It should still work the same as before as an independent package

**Comment by BaronVonPerko** [comment]
Just to note, I needed to also update my `vite` to version 5 (it was 4 and was throwing the ESM error)

**Comment by brandonroberts** [comment]
Good catch. Thanks @BaronVonPerko

**Comment by sonukapoor** [comment]
@brandonroberts I can look into this.
